secret_store/src/listener/http_listener.rs

```diff
@@ -68,83 +68,44 @@ enum Request {
 }
 
 /// Cloneable http handler
-struct KeyServerHttpHandler<T: KeyServer + 'static> {
-	handler: Arc<KeyServerSharedHttpHandler<T>>,
+struct KeyServerHttpHandler {
+	handler: Arc<KeyServerSharedHttpHandler>,
 }
 
 /// Shared http handler
-struct KeyServerSharedHttpHandler<T: KeyServer + 'static> {
-	key_server: T,
+struct KeyServerSharedHttpHandler {
+	key_server: Arc<KeyServer>,
 }
 
-impl<T> KeyServerHttpListener<T> where T: KeyServer + 'static {
+impl KeyServerHttpListener {
 	/// Start KeyServer http listener
-	pub fn start(listener_address: Option<NodeAddress>, key_server: T) -> Result<Self, Error> {
+	pub fn start(listener_address: NodeAddress, key_server: Arc<KeyServer>) -> Result<Self, Error> {
 		let shared_handler = Arc::new(KeyServerSharedHttpHandler {
 			key_server: key_server,
 		});
 
-		let http_server = listener_address
-			.map(|listener_address| format!("{}:{}", listener_address.address, listener_address.port))
-			.map(|listener_address| HttpServer::http(&listener_address).expect("cannot start HttpServer"))
-			.map(|http_server| http_server.handle(KeyServerHttpHandler {
+		let listener_address = format!("{}:{}", listener_address.address, listener_address.port);
+		let http_server = HttpServer::http(&listener_address).expect("cannot start HttpServer");
```

Comment:
Shouldn't this be converted to an error?

Reply:
Actually I've got this pattern from other parts of our code - there's a lot of expects like that when something crucial is **failed to initialize**. Like this:
https://github.com/paritytech/parity/blob/master/util/network/src/discovery.rs#L112
https://github.com/paritytech/parity/blob/master/util/reactor/src/lib.rs#L43
...

I do not know - what is better, though :) If you say - `Err` is better, please respond - I'll add the handling

Reply:
The second example is only spawning an event loop (afair it fails only in some extremely rare, low-level cases).
The first one seems similar, I'm opting for an error, since for instance the port can be taken already and that imho shouldn't lead to a crash but rather a more friendly message.